python/example_code/medical-imaging/medical_imaging_basics.py

```diff
@@ -441,4 +461,181 @@ def list_tags_for_resource(self, resource_arn):
+
+        datastores = self.list_datastores()
+        print(f"datastores : {datastores}")
+
+        job_name = "python_usage_demo_job"
+        job_id = self.start_dicom_import_job(job_name, data_store_id,
+                                             data_access_role_arn,
+                                             source_s3_uri, dest_s3_uri)
+        print(f"Started import job with id: {job_id}")
+
+        while True:
+            time.sleep(1)
+            job = self.get_dicom_import_job(data_store_id, job_id)
+            job_status = job['jobStatus']
+            print(f'Status of import job : "{job_status}"')
+            if job_status == "COMPLETED":
+                break
+            elif job_status == "FAILED":
+                raise Exception("DICOM import job failed")
+
+        import_jobs = self.list_dicom_import_jobs(data_store_id)
+        print(import_jobs)
+        for job in import_jobs:
+            print(job)
+
+            # Search with EQUAL operator..
+            # snippet-start:[python.example_code.medical-imaging.SearchImageSets.use_case1]
+        filter = {"filters": [{"operator": "EQUAL", "values": [{"DICOMPatientId": "3524578"}]}]}
+
+        image_sets = self.search_image_sets(data_store_id, filter)
+        # snippet-end:[python.example_code.medical-imaging.SearchImageSets.use_case1]
+
+        # Search with BETWEEN operator using DICOMStudyDate and DICOMStudyTime.
+        # snippet-start:[python.example_code.medical-imaging.SearchImageSets.use_case2]
+        filter = {"filters": [{"operator": "BETWEEN", "values": [
+            {"DICOMStudyDateAndTime": {"DICOMStudyDate": "19900101", "DICOMStudyTime": "000000"}},
+            {"DICOMStudyDateAndTime": {"DICOMStudyDate": "20230101", "DICOMStudyTime": "000000"}}]}]}
+
+        image_sets = self.search_image_sets(data_store_id, filter)
+        # snippet-end:[python.example_code.medical-imaging.SearchImageSets.use_case2]
+
+        # Search with BETWEEN operator using createdAt. Time studies were previously persisted.
+        # snippet-start:[python.example_code.medical-imaging.SearchImageSets.use_case3]
+        filter = {
+            "filters": [{
+                "values": [{"createdAt": datetime.datetime(2021, 8, 4, 14, 49, 54, 429000)},
+                           {"createdAt": datetime.datetime.now() + datetime.timedelta(days=1)}],
+                "operator": "BETWEEN"
+            }]
+        }
+
+        image_sets = self.search_image_sets(data_store_id, filter)
+        # snippet-end:[python.example_code.medical-imaging.SearchImageSets.use_case3]
+
+        image_set_ids = [image_set['imageSetId'] for image_set in image_sets]
+        for image_set in image_sets:
+            print(image_set)
+
+        image_set_id = image_sets[0]['imageSetId']
+        version_id = image_sets[0]['version']
+        returned_image_set = self.get_image_set(data_store_id, image_set_id, str(version_id))
+        print(returned_image_set)
```

Comment:
This works functionally, but I would put this into a method to make the code more readable and most importantly testable.